lesson34/src/js/mypage.js

```diff
@@ -0,0 +1,158 @@
+
+    favoriteContent
+      .appendChild(createFavoriteTitle(title))
+      .after(favoriteDate);
+    favoriteContent.append(createReleaseButton());
+    fragment
+      .appendChild(favoriteItem)
+      .appendChild(favoriteLink)
+      .appendChild(createFavoriteThumbnail(thumbnail))
+      .after(favoriteContent);
+  });
+  return fragment;
+};
+
+const createFavoriteTitle = (text) => {
+  const favoriteTitle = createElementWithClassName("p", "mypage__favorite-title");
+  const maxLength = 100;
+  let titleStr = "";
+  titleStr = text.length > maxLength ? text.substr(0, maxLength) + "..." : text;
+  favoriteTitle.textContent = titleStr;
+  return favoriteTitle;
+}
+
+const createFavoriteThumbnail = (imgData) => {
+  const favoriteThumbnailWrapper = createElementWithClassName("div", "mypage__favorite-img-wrapper");
+  const favoriteThumbnail = document.createElement("img");
+  favoriteThumbnail.src = imgData || "./img/no-image.jpeg";
+  favoriteThumbnailWrapper.appendChild(favoriteThumbnail);
+  return favoriteThumbnailWrapper;
+};
+
+const createReleaseButton = () => {
+  const releaseButton = createElementWithClassName("button", "mypage__favorite-button");
+  const starIcon = document.createElement("img")
+  releaseButton.classList.add("js-favorite-release-button");
+  starIcon.src = "./img/star-icon--disabled.svg";
+  releaseButton.textContent = "お気に入りから解除する";
+  releaseButton.appendChild(starIcon);
+  return releaseButton;
+};
+
+const removeClickedArticles = (favoriteArticles, target) => {
+  const articleId = target.closest("li").id;
+  return favoriteArticles.filter(article => !article[articleId]);
+}
+
+const isEmptyFavoriteArticles = (favoriteArticles) => Object.keys(favoriteArticles).length === 0;
+
+const changeFavoriteArticlesInLocalStorage = (target) => {
+  const favoriteArticles = JSON.parse(localStorage.getItem("favoriteArticles"));
+  const filteredFavoriteArticles = removeClickedArticles(favoriteArticles[userToken], target);
+
+  if (filteredFavoriteArticles.length === 0) {
+    delete favoriteArticles[userToken];
+  } else {
+    favoriteArticles[userToken] = filteredFavoriteArticles;
+  }
+
+  if (isEmptyFavoriteArticles(favoriteArticles)) {
+    localStorage.removeItem("favoriteArticles");
+    return;
+  }
+
+  localStorage.setItem("favoriteArticles", JSON.stringify(favoriteArticles));
+};
+
+const addEventListenerOfFavoriteReleaseButton = () => {
+  const releaseButtons = [...document.querySelectorAll(".js-favorite-release-button")];
+  releaseButtons.forEach((button) => {
+    button.addEventListener("click", (event) => {
+      event.preventDefault();
+      changeFavoriteArticlesInLocalStorage(event.target);
+      event.target.closest("li").remove();
+    });
+  });
+};
+
+const renderMessageOfNoFavoriteArticle = () => {
+  const noFavoriteArticle = createElementWithClassName("p", "no-favorite");
+  noFavoriteArticle.textContent = "お気に入りに登録されている記事はありません";
+  favoriteWrapper.append(noFavoriteArticle);
+}
```

Comment:
Thanks for the review.
It's hard to tell, isn't it...
Fixed it so that the element is checked for each release and if not, a message is displayed.
[c4de821](https://github.com/sae-github/JS_lesson/pull/43/commits/c4de8214d3d4eae0c1b25bfceac505d27e500d81)